Implicitless serialization

laughedelic opened the pull request:
See https://github.com/ohnosequences/blast-api/pull/15.
Also fixes #19.

laughedelic committed:
Split sources in several files; Structured code similarly to blast-api

laughedelic committed:
Added apply method to the flash command

laughedelic committed:
Fixed tests code

laughedelic commented:
Although being still overcomplicated, now this looks more like the code in blast-api (which has been maintained more actively). Also now it cross-compiles with Scala 2.11/2.12 and expression `toSeq` conversion doesn't require implicits.

@eparejatobes are you going to review it?

laughedelic committed:
Dropped read-len, fragment-len and fragment-len-stddev options

eparejatobes reviewed (commented):
I don't see much point in updating this project right now. We will stop using FLASH soon, and in the meantime better not to redo from scratch something that's working.

As long as it works, I'm fine with it. Merge it upon your responsibility.

laughedelic committed:
Merge pull request #21 from ohnosequences/21/drop-options

Remove several options

laughedelic commented:
Well, MG7 still depends on it. And I didn't redo it from scratch, just refactored it and changed (simplified) basically one thing: converting expression to `Seq[String]`. There's only one simple test here and it passes. I will test it in MG7, of course, too.
